src/lib.rs

```diff
@@ -40,188 +46,125 @@
 //!
 //!     Ok(())
 //! }
-//!
-//! fn init_tracer() -> thrift::Result<()> {
-//!     let exporter = opentelemetry_jaeger::Exporter::builder()
-//!         .with_agent_endpoint("127.0.0.1:6831".parse().expect("not a valid endpoint"))
-//!         .with_process(opentelemetry_jaeger::Process {
-//!             service_name: "example-server".into(),
-//!             tags: vec![KeyValue::new("exporter", "jaeger")],
-//!         })
-//!         .init()?;
-//!
-//!     let provider = sdk::Provider::builder()
-//!         .with_simple_exporter(exporter)
-//!         .with_config(sdk::Config {
-//!             default_sampler: Box::new(sdk::Sampler::AlwaysOn),
-//!             ..Default::default()
-//!         })
-//!         .build();
-//!     global::set_provider(provider);
-//!
-//!     Ok(())
-//! }
 //! ```
 
 #![forbid(unsafe_code)]
 #![cfg_attr(feature = "docs", feature(doc_cfg))]
 #![deny(missing_docs)]
 #![deny(unused_imports)]
 #![deny(missing_debug_implementations)]
+#![warn(clippy::expect_used)]
+#![deny(clippy::unwrap_used)]
+#![deny(unused_results)]
 #![doc(test(attr(allow(unused_variables), deny(warnings))))]
-#![allow(clippy::all)]
 
-use {
```

Comment:
I decided to unwrap the use statement because `rust-analyzer` isn't good at detecting them and prefers to insert new `use` statements. Happy to revert